incdec/src/incdec-plugin.c

```diff
@@ -0,0 +1,592 @@
+	{
+		plugin_data.show_in_menu = utils_get_setting_boolean(kf, CONF_GROUP, CONF_SHOW_IN_MENU, TRUE);
+	}
+
+	g_key_file_free(kf);
+	g_free(filename);
+}
+
+
+static gboolean save_config(void)
+{
+	GKeyFile *kf = g_key_file_new();
+	gchar *filename = get_config_filename();
+	gchar *dirname = g_path_get_dirname(filename);
+	gchar *data;
+	gsize length;
+	gboolean status;
+
+	g_key_file_set_boolean(kf, CONF_GROUP, CONF_SHOW_IN_MENU, plugin_data.show_in_menu);
+
+	utils_mkdir(dirname, TRUE);
+	data = g_key_file_to_data(kf, &length, NULL);
+	status = g_file_set_contents(filename, data, length, NULL);
+
+	g_free(data);
+	g_key_file_free(kf);
+	g_free(filename);
+	g_free(dirname);
+
+	return status;
+}
+
+
+static void on_configure_response(GtkDialog *dialog, gint response, gpointer user_data)
+{
+	if (response != GTK_RESPONSE_OK && response != GTK_RESPONSE_APPLY)
+		return;
+
+	plugin_data.show_in_menu = gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(plugin_data._display_in_popup));
+
+	if (! save_config())
+		dialogs_show_msgbox(GTK_MESSAGE_ERROR, _("Plugin configuration directory could not be saved."));
+
+	configuration_apply();
+}
+
+
+GtkWidget *plugin_configure(GtkDialog *dialog)
+{
+        GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 0);
+
+	plugin_data._display_in_popup = gtk_check_button_new_with_label(_("Display in Popup editor menu"));
+	gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(plugin_data._display_in_popup), plugin_data.show_in_menu);
+	gtk_box_pack_start(GTK_BOX(vbox), plugin_data._display_in_popup, FALSE, FALSE, 10);
+
+        gtk_widget_show_all(vbox);
+        g_signal_connect(dialog, "response", G_CALLBACK(on_configure_response), NULL);
+
+	return vbox;
+}
+
+
+void plugin_init (GeanyData *data)
+{
+	GeanyKeyGroup *key_group;
+
+	load_config();
+
+	key_group = plugin_set_key_group (geany_plugin, "incdec", KB_COUNT, on_change_number_callback);
+	keybindings_set_item (key_group, KB_INCREMENT_NUMBER, NULL, GDK_KEY_KP_Add, GDK_SHIFT_MASK,
+				"increment_number",
+				_("Increment Number By 1"), NULL);
+	keybindings_set_item (key_group, KB_DECREMENT_NUMBER, NULL, GDK_KEY_KP_Subtract, GDK_SHIFT_MASK,
+				"decrement_number",
+				_("Decrement Number By 1"), NULL);
+	keybindings_set_item (key_group, KB_INCREMENT_DECREMENT_NUMBER_X, NULL, GDK_KEY_KP_Multiply, GDK_SHIFT_MASK,
+				"increment_decrement_number_x",
+				_("Increment or Decrement Number X times"), NULL);
+
+	plugin_data._menu_item_sep = gtk_separator_menu_item_new();
+	gtk_container_add(GTK_CONTAINER(geany->main_widgets->editor_menu), plugin_data._menu_item_sep);
+
+	plugin_data._menu_item_change_number = gtk_menu_item_new_with_mnemonic(_("_Increment or Decrement number"));
+        gtk_container_add(GTK_CONTAINER(geany->main_widgets->editor_menu), plugin_data._menu_item_change_number);
+
+	configuration_apply();
+
+        g_signal_connect(plugin_data._menu_item_change_number, "activate", G_CALLBACK(on_change_number_x), NULL);
```

Comment:
Here and in some other places, there is mixed indentation, sometimes spaces sometimes tabs.
I suggest to use a consistent style.